saga/Repository.py

```diff
@@ -381,4 +385,113 @@ def _copy_file_to_dir(self, src, dst):
         if not os.path.isdir(dirname):
             os.makedirs(dirname)
 
-        shutil.copyfile(src, join(dst, src))
\ No newline at end of file
+        shutil.copyfile(src, join(dst, src))
+
+
+
+    def pull_folder(self, relative_folder_path):
+        # api-endpoint 
+        URL = "http://localhost:3000/get-folder"
+
+        base_directory = self.base_directory
+
+        shutil.rmtree(self.saga_directory)
+                
+        if not base_directory[len(base_directory) -1] == '/':
+            base_directory = base_directory + "/"
+        
+        try:
+            response = requests.get(url = URL, data={'folder_location' : relative_folder_path})
+            paths = response.json()
+            
+            # download all of the files
+            for path in paths['file_paths']:
```

Comment:
Why not make the returned paths object have: paths["file_paths"] and paths["folder_paths"]?

It seems more clear to me, than just "automatically" making folders that aren't empty when you make the files that contain them. 

Then, you could change the logic to:
1. first make all folders in paths["folder_paths"] 
2. make all the files in file_paths

Reply:
Feel free to push back if you disagree though!